docs/developer-guide/working-with-plugins.md

```diff
@@ -66,6 +66,28 @@ module.exports = {
 };
 ```
 
+### Custom Environments in Plugins
+
+Plugins can also provide custom environments. Custom environments can declare `globals`, turn `ecmaFeatures` on and off,
+and configure both core rules as well as rules included in the plugin. To provide custom environment, include `environment`
+property in your exported object.
```

Comment:
While you can't include `plugins` section, the thing that was bothering me, was that you can specify rules from another plugins. There's no validation currently anywhere in the code, to verify that rules inside environments don't include rules from some other plugin. I guess even if that happens, you are going to get a message about incorrect configuration of the rule, so not the end of the world.